src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+                    WriteUInt16((ushort)Registers.SD_CONFIG__INITIAL_PHASE_SD0, 0x0E0E);
+                    break;
+            }
+
+            SetTimingBudgetInMs(timingBudget);
+        }
+
+        /// <summary>
+        ///     This function returns the current distance mode (1 = Short, 2 = Long).
+        /// </summary>
+        /// <returns>The currently used distance mode.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">If reported distance mode is unknown</exception>
+        public DistanceMode GetDistanceMode()
+        {
+            byte temp = ReadByte((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP);
+            if (temp != 0x14 && temp != 0xA)
+            {
+                return DistanceMode.Unknown;
+            }
+
+            return temp == 0x14 ? DistanceMode.Short : DistanceMode.Long;
+        }
+
+        /// <summary>
+        ///     This function programs the intermeasurement period in ms.
+        ///     Intermeasurement period must be >/= timing budget. This condition is not checked by the API,
+        ///     the customer has the duty to check the condition. Default = 100 ms.
+        /// </summary>
+        /// <param name="interMeasMs">The inermeasurement period to set.</param>
+        public void SetInterMeasurementInMs(ushort interMeasMs)
+        {
+            uint clockPll = ReadUInt32((ushort)Registers.VL53L1X_RESULT__OSC_CALIBRATE_VAL);
+            clockPll &= 0x3FF;
+            WriteUInt32((ushort)Registers.VL53L1X_SYSTEM__INTERMEASUREMENT_PERIOD,
+                (uint)(clockPll * interMeasMs * 1.075));
+        }
+
+        /// <summary>
+        ///     This function returns the Intermeasurement period in ms.
+        /// </summary>
+        /// <returns>The Intermeasurement period in ms.</returns>
+        public ushort GetInterMeasurementInMs()
+        {
+            uint temp = ReadUInt32((ushort)Registers.VL53L1X_SYSTEM__INTERMEASUREMENT_PERIOD);
+            uint clockPll = ReadUInt32((ushort)Registers.VL53L1X_RESULT__OSC_CALIBRATE_VAL);
+
+            clockPll &= 0x3FF;
+
+            return (ushort)(temp / (clockPll * 1.065));
+        }
+
+        /// <summary>
+        ///     This function returns the <see cref="BootState" /> of the device (1 = booted, 0 = not booted).
+        /// </summary>
+        /// <returns>The <see cref="BootState" /> of the device.</returns>
+        public BootState GetBootState()
+        {
+            byte bootstateVal = ReadByte((ushort)Registers.VL53L1X_FIRMWARE__SYSTEM_STATUS);
+            return
+                bootstateVal > 0
+                    ? BootState.Booted
+                    : BootState.NotBooted; // (BootState)ReadByte((ushort)Registers.VL53L1X_FIRMWARE__SYSTEM_STATUS);
+        }
+
+        /// <summary>
+        ///     This function returns the sensor ID which must be 0xEEAC.
+        /// </summary>
+        /// <returns>The sensor id.</returns>
+        public ushort GetSensorId()
```

Comment:
From the sister device, you have an information class, maybe something you can reuse? https://github.com/dotnet/iot/blob/main/src/devices/Vl53L0X/Information.cs and the logic behind as well

Reply:
I wanted to implement something like this is the first place, but the only thing i found was the SensorID. If someone knows how to get more information out of the device, i'll be happy to implement it.